serialization/Inflater.hx

```diff
@@ -493,6 +525,130 @@ class Inflater {
     }
   }
 
+  public function inflateEnum() : InflatedEnum {
+    switch( stream.readByte() ) {
+    default:
+      throw "Invalid enum type";
+      return null;
+    case "-".code:
+      var t = readDigits();
+      if( t < 0 || t >= tcache.length )
+        throw "Invalid enum type reference";
+      if( stream.readByte() != ":".code )
+        throw "Invalid enum type reference format";
+      return tcache[t];
+    case "_".code:
+      return inflateEnumInfo();
+    }
+  }
+
+  function inflateEnumInfo() : InflatedEnum {
+    var info : InflatedEnum = new InflatedEnum();
+    info.name = unserialize();
+    if( stream.readByte() != ":".code ) {
+      throw "Invalid type format (cls)";
+    }
+    info.type = Type.resolveEnum(info.name);
```

Comment:
This should be a nice performance boost: Type.resolveEnum isn't fast, and we were previously calling it for every enum value.